src/mlpack/core/util/arma_traits.hpp

```diff
@@ -366,6 +366,12 @@ struct IsArma
                                 arma::is_arma_sparse_type<T>::value;
 };
 
+template<typename T>
+struct IsSparse
+{
+  constexpr static bool value = arma::is_arma_sparse_type<T>::value;
+};
+
```

Comment:
I have already added this one, I think it is called `IsSparseMat` somewhere in this file

Reply:
`IsSparse` is actually very slightly different---it returns true for any Armadillo sparse expression (so, e.g., even `sparse_matrix.col(3).t()` or something like this).  `IsSparseMat` will only return `true` when the given object is literally an `arma::SpMat`.  I have no objections to reorganization/renaming of all of these traits structures---we've got quite a lot of them at this point and I think they grew organically instead of in a planned manner :)